metadata/containers.go

```diff
@@ -71,6 +82,48 @@ func (s *containerStore) List(ctx context.Context, filter string) ([]containers.
 	return m, nil
 }
 
+func adaptContainer(o interface{}) filters.Adaptor {
+	obj := o.(containers.Container)
+	return filters.AdapterFunc(func(fieldpath []string) (string, bool) {
+		if len(fieldpath) == 0 {
+			return "", false
+		}
+
+		switch fieldpath[0] {
```

Comment:
This code can hurt the brain. :face_with_head_bandage: 

I _think_ this is right. I do want to figure out a better way to do this that doesn't require a complex closure for every object that is filtered but I have yet to benchmark this.